sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java

```diff
@@ -0,0 +1,500 @@
+                    for (HybridSearchQueryResult<Document> hybridSearchQueryResult : feedOfHybridSearchQueryResults.getResults()) {
+                        unwrappedResults.add(hybridSearchQueryResult.getPayload());
+                    }
+                    FeedResponse<Document> feedResponse = BridgeInternal.createFeedResponseWithQueryMetrics(unwrappedResults,
+                        feedOfHybridSearchQueryResults.getResponseHeaders(),
+                        BridgeInternal.queryMetricsFromFeedResponse(feedOfHybridSearchQueryResults),
+                        ModelBridgeInternal.getQueryPlanDiagnosticsContext(feedOfHybridSearchQueryResults),
+                        false,
+                        false,
+                        feedOfHybridSearchQueryResults.getCosmosDiagnostics());
+                    diagnosticsAccessor.addClientSideDiagnosticsToFeed(
+                        feedResponse.getCosmosDiagnostics(), clientSideRequestStatistics);
+                    return feedResponse;
+                }).switchIfEmpty(Flux.defer(() -> {
+                    FeedResponse<Document> frp = BridgeInternal.createFeedResponseWithQueryMetrics(Utils.immutableListOf(),
+                        headerResponse(
+                            tracker.getAndResetCharge()),
+                        queryMetricMap,
+                        null,
+                        false,
+                        false,
+                        null);
+                    diagnosticsAccessor.addClientSideDiagnosticsToFeed(
+                        frp.getCosmosDiagnostics(), clientSideRequestStatistics);
+                    return Flux.just(frp);
+            }));
+        }
+    }
+
+    private static Flux<HybridSearchQueryResult<Document>> computeRRFScores(Mono<List<List<Integer>>> ranks, Mono<List<HybridSearchQueryResult<Document>>> coalescedAndSortedResults) {
+        return ranks.zipWith(coalescedAndSortedResults)
+            .map(tuple -> {
+                List<List<Integer>> ranksInternal = tuple.getT1();
+                List<HybridSearchQueryResult<Document>> results = tuple.getT2();
+
+                for (int index = 0; index < results.size(); ++index) {
+                    double rrfScore = 0.0;
+                    for (List<Integer> integers : ranksInternal) {
+                        rrfScore += 1.0 / (RRF_CONSTANT + integers.get(index));
+                    }
+
+                    results.get(index).setScore(rrfScore);
+                }
+                // Sort on the RRF scores to build the final result
+                results.sort(Comparator.comparing(HybridSearchQueryResult::getScore, Comparator.reverseOrder()));
+                return results;
+            }).flatMapMany(Flux::fromIterable);
+    }
+
+    private static Mono<List<List<Integer>>> computeRanks(Mono<List<List<ScoreTuple>>> componentScoresList) {
+        return componentScoresList.flatMap(componentScores -> {
```

Comment:
Please define concurrency for flatMap explicitly

Reply:
I am doing a flatMap on Mono, which doesn't takes in a concurrency parameter. And going over the code again we don't need a flatMap here, so I have updated the code to use a map.